src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerExecContainerFunctionalTest.groovy

```diff
@@ -40,6 +40,32 @@ class DockerExecContainerFunctionalTest extends AbstractGroovyDslFunctionalTest
         result.output.contains('Hello World')
     }
 
+    def "Execute custom commands within running container"() {
```

Comment:
How exactly does this fix the issue you reported?

Reply:
The `DockerExecContainer.doRunRemoteCommand` method is both iterating over the `commands` _and_ running them, so if you want to extend the task to write your own logic to determine what the commands are (like I want to do so I lazily wrap some around a new input property) you need to duplicate all that logic.

There's no bug or change of behavior, I've just moved the logic that iterates over the commands out so you can override it in isolation and avoid all the duplication. The test doesn't do what I'm interested in doing, its just demonstrating you can override the task to write whatever you want to determine your commands (if I can get it running and fix any more booboos I made).